tests/trilinos/direct_solver_2.cc

```diff
@@ -0,0 +1,334 @@
+  const unsigned int   n_q_points    = quadrature_formula.size();
+
+  FullMatrix<double>   cell_matrix (dofs_per_cell, dofs_per_cell);
+  Vector<double>       cell_rhs (dofs_per_cell);
+
+  std::vector<types::global_dof_index> local_dof_indices (dofs_per_cell);
+
+  typename DoFHandler<dim>::active_cell_iterator
+  cell = dof_handler.begin_active(),
+  endc = dof_handler.end();
+
+  for (; cell!=endc; ++cell)
+    {
+      fe_values.reinit (cell);
+      cell_matrix = 0;
+      cell_rhs = 0;
+
+      for (unsigned int q_point=0; q_point<n_q_points; ++q_point)
+        for (unsigned int i=0; i<dofs_per_cell; ++i)
+          {
+            for (unsigned int j=0; j<dofs_per_cell; ++j)
+              cell_matrix(i,j) += (fe_values.shape_grad (i, q_point) *
+                                   fe_values.shape_grad (j, q_point) *
+                                   fe_values.JxW (q_point));
+
+            cell_rhs(i) += (fe_values.shape_value (i, q_point) *
+                            right_hand_side.value (fe_values.quadrature_point (q_point)) *
+                            fe_values.JxW (q_point));
+          }
+
+      cell->get_dof_indices (local_dof_indices);
+      constraints.distribute_local_to_global(cell_matrix, cell_rhs,
+                                             local_dof_indices,
+                                             system_matrix, system_rhs);
+    }
+  system_matrix.compress(VectorOperation::add);
+}
+
+
+
+template <int dim>
+void Step4<dim>::solve ()
+{
+  // Compute 'reference' solution with CG solver and SSOR preconditioner
+  TrilinosWrappers::PreconditionSSOR preconditioner;
+  solution = 0;
+  SolverControl           solver_control (1000, 1e-12);
+  SolverCG<>              solver (solver_control);
+  preconditioner.initialize(system_matrix);
+  solver.solve (system_matrix, solution, system_rhs,
+                preconditioner);
+
+  Vector<double> output(solution);
+  {
+    deallog.push("DirectKLU");
+    TrilinosWrappers::SolverDirect::AdditionalData data;
+    data.solver_type = "Amesos_Klu";
+    SolverControl           solver_control (1000, 1e-10);
+    TrilinosWrappers::SolverDirect solver(solver_control, data);
+    solver.initialize (system_matrix);
+    solver.solve (system_matrix, output, system_rhs);
+    output -= solution;
+    deallog << "Norm of error in direct solve: " << output.l2_norm()
+            << std::endl;
+    deallog.pop();
+  }
+
+  {
+    deallog.push("DirectLAPACK");
+    TrilinosWrappers::SolverDirect::AdditionalData data;
+    data.solver_type = "Amesos_Lapack";
+    SolverControl           solver_control (1000, 1e-10);
+    TrilinosWrappers::SolverDirect solver(solver_control, data);
+    solver.initialize (system_matrix);
+    solver.solve (output, system_rhs);
+    output -= solution;
+    deallog << "Norm of error in direct solve: " << output.l2_norm()
+            << std::endl;
+    deallog.pop();
+  }
+
+  {
+    deallog.push("DirectDscpack");
```

Comment:
from https://trilinos.org/packages/amesos/amesos-installation/amesos-dscpack-install/ `Dscpack` needs `superlu`, most likely your `trilinos` installation is not configured with it. I would suggest leaving only `DirectLAPACK` and `DirectKLU` as the unit test is not about the variety of solvers but the ability to apply same factorisation to different RHS's.